File: framework/ui-elements/src/displays/components/Item.js
import React from "react";
import AbstractItem from "../../../gen/displays/components/AbstractItem";
import ItemNotifier from "../../../gen/displays/notifiers/ItemNotifier";
import ItemRequester from "../../../gen/displays/requesters/ItemRequester";
import DisplayFactory from "alexandria-ui-elements/src/displays/DisplayFactory";

export default class Item extends AbstractItem {

	constructor(props) {
		super(props);
		this.notifier = new ItemNotifier(this);
		this.requester = new ItemRequester(this);
	};

}

DisplayFactory.register("Item", Item);

File: framework/ui/src/io/intino/alexandria/ui/displays/components/Item.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.components.collection.CollectionItemComponent;
import io.intino.alexandria.ui.displays.notifiers.ItemNotifier;

public class Item<DN extends ItemNotifier, Type, B extends Box> extends CollectionItemComponent<DN, Type, B> {
    private Type item;

    public Item(B box) {
        super(box);
    }

    @Override
    public Type item() {
        return this.item;
    }

    @Override
    public void update(Type item) {
        item(item);
    }

    public Item<DN, Type, B> item(Type item) {
        this.item = item;
        return this;
    }

}